### PR TITLE
Small improvements to searchspaces and simulation mode

### PR DESCRIPTION
**Pull request by fjwillemsen**
This pull request solves various small issues regarding search spaces and simulation mode:

- The constraint parser now works correctly when round brackets surround a restriction. 
- Fixed an issue where the automatically added check that the product of `block_sizes` does not exceed `max_threads` was not added back to the restrictions, which were later rechecked for each configuration if `check_restrictions` is enabled, causing a discrepancy. 
- Constraints can now also be evaluated from a tuple combination by the `check_restrictions` function.
- An empty tuple is added for unknown parameters of ignored restrictions. 
- The `MinProd` constraint was added to the conversion function for backwards compatibility.

### CODE REVIEW

File: kernel_tuner/util.py [hunk 1]
```diff
@@ -102,9 +102,9 @@ class StopCriterionReached(Exception):
     "block_size_x",
     "block_size_y",
     "block_size_z",
-    "ngangs",
-    "nworkers",
-    "vlength",
+    # "ngangs",
+    # "nworkers",
+    # "vlength",
```
Comment (benvanwerkhoven) [inline kernel_tuner/util.py:107]: Why are these lines commented?

File: kernel_tuner/util.py [hunk 2]
```diff
@@ -284,9 +284,15 @@ def check_restrictions(restrictions, params: dict, verbose: bool) -> bool:
...
-                    break
+                if isinstance(restrict, Constraint):
+                    restrict = convert_constraint_restriction(restrict)
+                    if not restrict(list(selected_params.values())):
+                        valid = False
+                        break
+                else:
+                    if not restrict(**selected_params):
+                        valid = False
+                        break
```
Comment (benvanwerkhoven) [inline kernel_tuner/util.py:295]: It seems our automated linting system 'SonarCloud' isn't too happy about this level of nesting control flow. Maybe we can break things up into smaller chunks?

Comment (fjwillemsen) [reply kernel_tuner/util.py]: I've refactored this into two separate functions, SonarCloud is happy again.

File: kernel_tuner/util.py (hunk 1 above)
Comment (fjwillemsen) [reply kernel_tuner/util.py:107]: These block size names were added for OpenACC. The default block sizes were added automatically if the block size is not defined by the user, causing issues with these parameters outside of OpenACC. I spoke with @isazi about this, if I understood correctly these are no longer needed, hence commented pending removal.

**Comment by benvanwerkhoven** [comment]
Thanks!

**Pull request merged**